### DISCUSSION TITLE
How do I find out the IP address of my peer?

### DISCUSSION THREAD

**Discussion by marek22k** [socketry/async-io#62] [Q&A] [answered]
Hello,

I want to run a Whois server with Async and for debug purposes provide the possibility to output the IP address of the client. How can I find out this? I have an Async Peer and an Async Stream object - how can I find out the IP address of the client?

**Comment by ioquatix** [comment, marked as answer]
https://rubyapi.org/3.2/o/basicsocket#method-i-remote_address should help you.

**Comment by marek22k** [comment]
`peer.remote_address` works. Thanks!
